### ISSUE TITLE
R# Reports all tests as Inconclusive for project with NUnit 3.12 + NUnit3TestAdapter 3.13 installed

### ISSUE THREAD

**Issue by stevenaw** [nunit/nunit3-vs-adapter#638]
A coworker encountered this today while we were updating test dependencies. All our testing was on Visual Studio 2017, for projects targeting .NET 462.

It seems like, when running tests in R# for a test project with NuGet packages `NUnit 3.12` + `NUnit3TestAdapter 3.13` installed, R# reports the test status of all tests as inconclusive. We didn't see this behaviour with `NUnit 3.11.2` + `NUnitTestAdapter 3.13`, or with `NUnit 3.12` and no test adapter installed as a NuGet package. It also worked as expected when using the latest version of the VS Extension Test Adapter with `NUnit 3.12` + `NUnit3TestAdapter 3.13` installed as NuGet packages.

Filing this here because it seems adapter-centric to me.
I don't have specifics on exact versions of R# or VS 2017 used, but VS 2017 is `15.9.*`.

**Comment by luisdaglez** [comment]
@stevenaw you beat me to it by a couple minutes. 
Reproducible on a brand new from scratch solution.
Visual Studio 15.9.13
ReSharper Ultimate 2018.1

**Comment by OsirisTerje** [comment]
I can reproduce this, so thanks for reporting @stevenaw  and @luisdaglez .  
(And I am using VS2017 15.9.9, with Resharper 2017.3.) 
I also tried to change the .net framework to 4.7.2, but same result. 
Dropping the adapter down to 3.11.2 resolves the issue.  
  
There should be no changes in the adapter that would give this result, so I am unsure about what this is. 
It would be good if someone from JetBrains could have a first look at this.  Can one of you guys raise an issue at their site, and refer back to this issue ?

**Comment by jnm2** [comment]
Does it make any difference whether https://www.nuget.org/packages/Microsoft.NET.Test.Sdk/ is referenced (per VSTest guidelines)?

**Comment by jnm2** [comment]
I can't repro with ReSharper 2019.1.1 on Visual Studio 2019.1.6, Class1.cs as above and this csproj:

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>net472</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="NUnit" Version="3.12.0" />
    <PackageReference Include="NUnit3TestAdapter" Version="3.12.0" />
  </ItemGroup>

</Project>
```

**Comment by jnm2** [comment]
I can't repro with ReSharper 2019.1.3, either.

**Comment by jnm2** [comment]
@OsirisTerje Is your repro using the legacy csproj format or packages.config? Maybe that's related somehow, e.g. packages.config doesn't do transitive package references?

**Comment by OsirisTerje** [comment]
@jnm2 I am using the  legacy project format with a packages.config file.

**Comment by stevenaw** [comment]
Thanks for following up on this @osiristerje and @jnm2. We didn't have the chance to check the Microsoft Test SDK thing (hopefully Monday), but I can confirm we were also using the legacy project format and packages.config.

I'll give the new format a test this week using our same R# and VS versions, and see what happens.

**Comment by OsirisTerje** [comment]
@stevenaw Thanks!  As I mentioned above, I could not make the new format work.  Good if you can verify. 
Also, I don't have VS2019 on my "vacation laptop", and no high speed line to install it either.  @jnm2 says it works with VS 2019, so if we can get confirmation on that too, it would be helpful. 
It would also be very helpful if you could raise an issue on JetBrains Resharper support, and link to this one.

**Comment by stevenaw** [comment]
Aha. Interesting. It works for me with the legacy project format. However, it looks like with the following setup, the bug occurs on `R# 2018.1` and not on `R# 2018.3`:
- NUnit 3.12 (current)
- NUnit3Adapter 3.13 (current)
- Test Project = NET 462
- Project File = Legacy Format

When target the same NUnit versions on the new project format (+ .NET Core 2.1) it also all passed on `R# 2018.3`.

@OsirisTerje Sorry but it took me a bit to find JetBrains's issue tracker. Now that I've found it, I think I've found an existing (and since fixed) issue for this there: https://youtrack.jetbrains.com/issue/RSRP-469301

I haven't made another issue there to track it (but can if you still think it's useful). Though as you probably already suspected, this sounds like it is an issue on the R# side.
